frontend/src/components/form/ApiWrapper.vue

```diff
@@ -0,0 +1,164 @@
+<!--
+Wrapper component for form components to save data back to API
+-->
+
+<template>
+  <v-form
+    inline
+    class="mb-2">
+    <slot
+      :localValue="localValue"
+      :errorMessage="errorMessage"
+      :isSaving="isSaving"
+      :status="status"
+      :on="eventHandlers" />
+
+    <v-btn
+      v-if="!autoSave && !readonly"
+      :disabled="disabled"
+      small
+      color="warning"
+      class="mb-1"
+      @click="reset">
+      Reset
+    </v-btn>
+
+    <v-btn
+      v-if="!autoSave && !readonly"
+      color="primary"
+      small
+      :disabled="disabled || isSaving || (required && $v.localValue.$invalid)"
+      class="mr-2 ml-2 mb-1"
+      @click="save">
+      <v-progress-circular
+        v-if="isSaving"
+        indeterminate
+        color="primary"
+        size="20"
+        class="mr-2" />
+
+      <v-icon
+        v-if="showSuccessIcon"
+        left>
+        mdi-check
+      </v-icon>
+
+      Save
+    </v-btn>
+  </v-form>
+</template>
+
+<script>
+
+import { debounce } from 'lodash'
+import { validationMixin } from 'vuelidate'
+import { required } from 'vuelidate/lib/validators'
+import { apiPropsMixin } from '@/mixins/apiPropsMixin'
+
+export default {
+  name: 'ApiWrapper',
+  mixins: [apiPropsMixin, validationMixin],
+  data () {
+    return {
+      localValue: this.value,
+      isSaving: false,
+      showSuccessIcon: false,
+      dirty: false,
+      eventHandlers: {
+        save: this.save,
+        reset: this.reset,
+        input: this.onInput,
+        touch: this.touch
+      }
+    }
+  },
+  validations: {
+    localValue: {
+      required
+    }
+  },
+  computed: {
+    isDirty: function () {
+      return this.value !== this.localValue
+    },
+    errorMessage () {
+      const errors = []
+      if (!this.$v.localValue.$dirty) return errors
+      !this.$v.localValue.required && errors.push('Feld darf nicht leer sein.')
+      return errors
+    },
+    status: function () {
+      if (this.isSaving) {
+        return 'saving'
+      } else if (this.showSuccessIcon) {
+        return 'success'
+      } else {
+        return 'init'
+      }
+    }
+  },
+  watch: {
+    value: function (newValue, oldValue) {
+      // override local value if it wasn't dirty
+      if (!this.dirty || this.overrideDirty) {
+        this.localValue = newValue
+      }
+
+      // clear dirty if outside value changes to same as local value (e.g. after save operation)
+      if (this.localValue === newValue) {
+        this.dirty = false
+      }
+    }
+  },
+  created: function () {
+    // Create debounced save method (lodash debounce function)
+    this.debouncedSave = debounce(this.save, this.autoSaveDelay)
```

Comment:
Ist das wirklich die korrekte Art, das zu machen? Setzt das nicht debouncedSave auf der `data` der Komponente ohne Vue davon zu erzählen? Könnten wir das stattdessen als `method` realisieren? https://alligator.io/vuejs/lodash-throttle-debounce#debouncing-methods
Im Vue-Mindset so wie ich es verstehe sollte so viel wie möglich deklarativ gemacht werden und die imperativen `created`, `mounted` und `watch` so wenig wie möglich verwendet werden.

Reply:
Yep, ist definitiv cleaner so

Reply:
War nun etwas "tricky": ich kann `debounceSave` nicht in die `methods` aufnehmen wie im Link beschrieben, weil da hab ich keinen Zugriff auf `this.autoSaveDelay`. Lösung ist, `debouncedSave` als computed method zu deklarieren. 

Dann könnte man theoretisch sogar dynamisch `autoSaveDelay` ändern. Nicht, dass ich das wollte, aber now you can :-)